Update couch docker image to apache/couchdb:3.3.1

skodde opened the pull request:
## Technical Summary
https://dimagi-dev.atlassian.net/browse/SAAS-14491

## Safety Assurance

### Safety story
This only concerns a dev/local setup.

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

skodde committed:
Update couch docker image to apache/couchdb:3.3.1

millerdev commented on dev_settings.py:
```diff
@@ -82,8 +82,8 @@
-        'COUCH_USERNAME': '',
-        'COUCH_PASSWORD': '',
+        'COUCH_USERNAME': 'admin',
+        'COUCH_PASSWORD': 'commcarehq',
```
Will this change break a local instance of HQ if the docker container is not also upgraded?

skodde replied:
It will. I wasn't able to find a solution for this. Every request made with these (or any) credentials to the current container will result in `error: unauthorized`.

Regarding your previous question, I just added a commit that should allow us to upgrade and downgrade the container without worrying about the local data, which should remain usable across versions without the need of any special operation.

dannyroberts replied:
My mental read of this is that as long as people are using the `localsettings.example.py` template which imports `*` from `dev_settings.py` and their local docker containers are updated right away, I imagine no one will have to do anything extra like manually edit any settings files or anything

millerdev replied:
It seems likely that devs will receive this change while the previous version of the docker container is still running. Any new Python  process (or existing Django web server process after settings are reloaded) that tries to access Couch after that point will fail with an authentication error until the new Couch container is started. The resolution to that error may not be obvious, so we may want to communicate to the team about how to resolve it.

millerdev replied:
> their local docker containers are updated right away

I don't think this is automatic. It will require something like `./scripts/docker up -d couch`

millerdev reviewed (commented):
Are there any compatibility constraints regarding on-disk data? Like is it possible to use the new container without migrating data from the old version? Similarly, is it possible to downgrade to the old container and continue with data written by 3.3.1?

dannyroberts commented:
Yeah I asked @skodde to PR just this minimal changeset in order to run tests against it, since that was an important piece of due diligence that was blocking the Ubuntu upgrade. He's working on the migration pathway for local dev setup now, and I'd support keeping this PR unmerged until that part of it is finalized as well

skodde committed:
Add custom node name to couch docker image

To be able to access data from the previous container, without
additional migration procedures, we need to use the same node name,
adding the '-name' parameter to the vm.args configuration file.

skodde commented:
@dimagi/dimagi-dev remember to run `./scripts/docker up -d couch` after merging this, and to check for `COUCH_DATABASES` definitions in your `localsettings.py`, which should be removed or updated with the new credentials present in `dev_settings.py` if needed. Otherwise, `dev_settings.py` is already automatically included and no action is needed.
